### PR TITLE
CI: add check-build for emulator

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
~~It's a draft.~~

### CODE REVIEW

File: .github/workflows/test-emulator.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+jobs:
+  build-EMULATOR:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout repository and submodules
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Cache pip
```
Comment (simonmicro) [inline .github/workflows/test-emulator.yml:16]: Not used - the emulator is on C++

File: .github/workflows/test-emulator.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+    - name: Cache PlatformIO
```
Comment (simonmicro) [inline .github/workflows/test-emulator.yml:23]: And the emulator does not use PlatformIO

File: .github/workflows/test-emulator.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ~/.platformio
+        key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+    - name: Install swig
```
Comment (simonmicro) [inline .github/workflows/test-emulator.yml:28]: The emulator does not support LUA (currently?)

**Review by simonmicro — CHANGES_REQUESTED** [review]

**Comment by simonmicro** [comment]
You write it is a draft -> please mark it as such!

**Review by RuffaloLavoisier — COMMENTED** [review]
@simonmicro LGTM

**Comment by simonmicro** [comment]
Superseeded by #289 289

**Pull request closed without merging**